### PR TITLE
Mesh area conservation

### PR DESCRIPTION
**Pull request by SchoeniPhlippsn**
## Description

This adds two area constraint potentials which can be applied to the mesh data structure.

## Motivation and context

These area constraints allow applying a global target area per mesh type or a local target area per mesh triangle to membranes represented by a particle mesh.

## How has this been tested?

I added pytests and used this potential in my current studies.

## Change log

```
Addition of area constraint mesh potentials
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by SchoeniPhlippsn** [comment]
pre-commit.ci autofix

**Comment by SchoeniPhlippsn** [comment]
pre-commit.ci autofix

**Comment by SchoeniPhlippsn** [comment]
pre-commit.ci autofix

**Comment by SchoeniPhlippsn** [comment]
@joaander @tcmoore3 
Ready for review!

**Comment by SchoeniPhlippsn** [comment]
pre-commit.ci autofix

File: hoomd/md/TriangleAreaConservationMeshForceCompute.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
```
Comment (joaander) [inline hoomd/md/TriangleAreaConservationMeshForceCompute.h]:
```suggestion
```
Duplicate line.

File: hoomd/md/TriangleAreaConservationMeshForceComputeGPU.cu [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+#include "TriangleAreaConservationMeshForceComputeGPU.cuh"
+#include "hip/hip_runtime.h"
+#include "hoomd/TextureTools.h"
+
+#include <assert.h>
+
+// SMALL a relatively small number
+#define SMALL Scalar(0.001)
+
```
Comment (joaander) [inline hoomd/md/TriangleAreaConservationMeshForceComputeGPU.cu]:
```suggestion
```
unused.

File: hoomd/md/TriangleAreaConservationMeshForceComputeGPU.cu [hunk 3]
```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2009-2024 The Regents of the University of Michigan.
+// Part of HOOMD-blue, released under the BSD 3-Clause License.
+
+#include "TriangleAreaConservationMeshForceComputeGPU.cuh"
+#include "hip/hip_runtime.h"
+#include "hoomd/TextureTools.h"
+
+#include <assert.h>
```
Comment (joaander) [inline hoomd/md/TriangleAreaConservationMeshForceComputeGPU.cu]:
```suggestion
```
unused.

**Review by joaander — CHANGES_REQUESTED** [review]
Thanks! Now that volume conservation is merged, you need to merge `trunk-minor` into this branch and resolve conflicts.

The code looks great. I found a few duplicate or unused lines of code that can be removed.

**Comment by SchoeniPhlippsn** [comment]
pre-commit.ci autofix

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**